docs/coding-guidelines/source-generator-guidelines.md

```diff
@@ -0,0 +1,27 @@
+Recommended reading to better understand source generators,
+[Roslyn Source Generators Cookbook](https://github.com/dotnet/roslyn/blob/main/docs/features/source-generators.cookbook.md).
+
+[Project guidance](./project-guidelines.md#directory-layout)
+
+[Packaging guidance](./libraries-packaging.md#analyzers--source-generators)
+
+## Source Generator Best Practices
+
+### DOs
+
+- **DO** generate code that looks as if a developer would write it manually.
+- **DO** emit strings rather than using the Roslyn Syntax API for better performance.
+- **DO** use consistent indentation and formatting in generated code.
+- **DO** make generators incremental whenever possible using the [`IIncrementalGenerator`](https://learn.microsoft.com/dotnet/api/microsoft.codeanalysis.iincrementalgenerator) interface.
```

Comment:
Not sure about this one. This should not be a "try to make your generators incremental if possible", but more a "make your generators properly incremental or don't ship them at all, because the alternative is that you'll murder the IDE".

Reply:
We should add an even stronger recommendation here against using `CreateSyntaxProvider` unless the generator is implementing an interceptor (there's no "accelerator" function for that scenario today).